### PR TITLE
Java Implementation of Kayaking 

### PR DESCRIPTION
**Pull request by brad182**

### CODE REVIEW

File: solutions/cf-863B.mdx [hunk 1]
```diff
@@ -48,5 +48,59 @@ int main() {
...
+
+		int[] W = new int[N];
+
+		for (int x = 0; x < W.length; x++) {
+			W[x] = io.nextInt();
+		}
+
+		Arrays.sort(W);
+
+		int minInstability = 2000000000;
```
Comment (nwatx) [inline solutions/cf-863B.mdx:73]:
```suggestion
		int minInstability = Integer.MAX_VALUE;
```

Comment (nwatx) [reply solutions/cf-863B.mdx:73]: is this a magic number?

**Review by nwatx — CHANGES_REQUESTED** [review]

File: solutions/cf-863B.mdx [hunk 2]
```diff
@@ -48,5 +48,59 @@ int main() {
...
+```java
+import java.util.*;
+import java.io.*;
+
+public class Kayaking {
+	public static void main (String[] args) {
+		Kattio io = new Kattio();
+
+		int N = io.nextInt();
+		N *= 2;
```
Comment (SansPapyrus683) [inline solutions/cf-863B.mdx:63]: replace these 2 lines with `int N = io.nextInt() * 2;`

File: solutions/cf-863B.mdx [hunk 3]
```diff
@@ -48,5 +48,59 @@ int main() {
...
+			W[x] = io.nextInt();
+		}
+
+		Arrays.sort(W);
+
+		int minInstability = 2000000000;
+
+		for (int i = 0; i < N; i++) {
+			for (int j = i + 1; j < N; j++) {
+				ArrayList<Integer> a = new ArrayList<>();
```
Comment (SansPapyrus683) [inline solutions/cf-863B.mdx:77]: perhaps a better variable name?

File: solutions/cf-863B.mdx [hunk 4]
```diff
@@ -48,5 +48,59 @@ int main() {
...
+
+				for (int k = 0; k < (N - 2); k+= 2) {
+					totalInstability += (a.get(k + 1)  - a.get(k));
+				}
+
+				minInstability = Integer.min(minInstability, totalInstability);
+			}
+		}
+
+		System.out.println(minInstability);
```
Comment (SansPapyrus683) [inline solutions/cf-863B.mdx:95]:
```suggestion
		io.println(minInstability);
```

File: solutions/cf-863B.mdx [hunk 5]
```diff
@@ -48,5 +48,59 @@ int main() {
...
+				for (int k = 0; k < N; k++) {
+					if ((k != i) && (k != j)) {
+						a.add(W[k]);
+					}
+				}
+
+				int totalInstability = 0;
+
+				for (int k = 0; k < (N - 2); k+= 2) {
+					totalInstability += (a.get(k + 1)  - a.get(k));
```
Comment (SansPapyrus683) [inline solutions/cf-863B.mdx:88]:
```suggestion
					totalInstability += a.get(k + 1)  - a.get(k);
```

File: solutions/cf-863B.mdx [hunk 6]
```diff
@@ -48,5 +48,59 @@ int main() {
...
+
+				minInstability = Integer.min(minInstability, totalInstability);
+			}
+		}
+
+		System.out.println(minInstability);
+		io.close();
+	}
+
+	//CodeSnip{Kattio}
```
Comment (SansPapyrus683) [inline solutions/cf-863B.mdx:99]: lines 61, 66, 70, 74, 86, and 90 are unnecessary blank lines

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cf-863B.mdx (hunk 6 above)
Comment (brad182) [reply solutions/cf-863B.mdx:99]: to me it looks cleaner and more readable with the blank lines

Comment (SansPapyrus683) [reply solutions/cf-863B.mdx:98]: it really doesn't, it just makes the code unnecessarily spaced out

File: solutions/cf-863B.mdx [hunk 7]
```diff
@@ -48,5 +48,58 @@ int main() {
...
+
+				for (int k = 0; k < N; k++) {
+					if ((k != i) && (k != j)) {
+						list.add(W[k]);
+					}
+				}
+
+				int totalInstability = 0;
+
+				for (int k = 0; k < (N - 2); k+= 2) {
```
Comment (dongliu0426) [inline solutions/cf-863B.mdx:86]: `k += 2`

File: solutions/cf-863B.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/cf-863B.mdx:98]: not resolved?

File: solutions/cf-863B.mdx [hunk 8]
```diff
@@ -48,5 +48,45 @@ int main() {
...
+		int N = io.nextInt() * 2;
+		int[] W = new int[N];
+		for (int x = 0; x < W.length; x++) {
+			W[x] = io.nextInt();
+		}
+		Arrays.sort(W);
+		int minInstability = Integer.MAX_VALUE;
+		for (int i = 0; i < N; i++) {
+			for (int j = i + 1; j < N; j++) {
+				ArrayList<Integer> list = new ArrayList<>();
```
Comment (dongliu0426) [inline solutions/cf-863B.mdx:70]: idk much about java but [Change to `List<>`](https://usaco.guide/general/adding-solution?lang=java#java)?

@HarryWangATX

Comment (nwatx) [reply solutions/cf-863B.mdx:70]: [yep](https://stackoverflow.com/questions/2279030/type-list-vs-type-arraylist-in-java)

**Review by nwatx — APPROVED** [review]

File: solutions/cf-863B.mdx (hunk 8 above)
Comment (HarryWangATX) [reply solutions/cf-863B.mdx:70]: ye

Comment (HarryWangATX) [reply solutions/cf-863B.mdx:70]: https://usaco.guide/general/adding-solution/#general

**Review by dongliu0426 — APPROVED** [review]

File: solutions/cf-863B.mdx [hunk 9]
```diff
@@ -2,19 +2,19 @@
...
 title: Kayaking
-author: Danh Ta Chi Thanh
+author: Danh Ta Chi Thanh, Brad Ma
 ---
 
 [Official Editorial](https://codeforces.com/blog/entry/54708)
 
 **Time Complexity:** $\mathcal{O}(N^3)$
 
+## Implementation
```
Comment (HarryWangATX) [inline solutions/cf-863B.mdx:12]: above time complexity

File: solutions/cf-863B.mdx [hunk 10]
```diff
@@ -2,19 +2,19 @@
 id: cf-863B
 source: CF
 title: Kayaking
-author: Danh Ta Chi Thanh
+author: Danh Ta Chi Thanh, Brad Ma
 ---
 
 [Official Editorial](https://codeforces.com/blog/entry/54708)
```
Comment (HarryWangATX) [inline solutions/cf-863B.mdx:8]:
```suggestion
[Official Analysis](https://codeforces.com/blog/entry/54708)
```

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: solutions/cf-863B.mdx (hunk 9 above)
Comment (brad182) [reply solutions/cf-863B.mdx:12]: the time complexity is there i'm not sure what you mean

Comment (dongliu0426) [reply solutions/cf-863B.mdx:12]: he meant to put this above time complexity

**Review by HarryWangATX — APPROVED** [review]

**Pull request merged**